frame/composable-traits/src/dex.rs

```diff
@@ -12,16 +12,37 @@ pub trait CurveAmm {
 	type Balance;
 	/// The user account identifier type for the runtime
 	type AccountId;
-	/// Type that represents index type of token in the pool passed from the outside as an extrinsic
-	/// argument.
-	type PoolTokenIndex;
 
 	/// Type that represents pool id
 	type PoolId;
 
 	/// Current number of pools (also ID for the next created pool)
 	fn pool_count() -> Self::PoolId;
 
+	/// Get pure exchange value for given units of given asset. (Note this does not include fees.)
+	fn get_exchange_value(
+		pool_id: Self::PoolId,
+		asset_id: Self::AssetId,
+		amount: Self::Balance,
+	) -> Result<Self::Balance, DispatchError>;
+
+	/// Buy given units of given asset from the pool.
+	/// In buy user does not know how much assets he/she has to exchange to get desired amount.
```

Comment:
/// `who` desires to buy `asset_id` of amount from `pool_id` withing given slippage `configuration`

so it lacks configuration parameter as for me.

Reply:
```
fn buy(
			who: &Self::AccountId,
			pool_id: Self::PoolId,
			asset_id: Self::AssetId,
			amount: Self::Balance,
		) -> Result<Self::Balance, DispatchError> {
			let assets_pair = PoolAssets::<T>::get(pool_id).ok_or(Error::<T>::PoolNotFound)?;
			let mut exchange_asset = assets_pair.quote;
			if asset_id == assets_pair.quote {
				exchange_asset = assets_pair.base;
			} else if asset_id != assets_pair.base {
				return Err(Error::<T>::AssetNotFound.into())
			}
			// Since when buying asset user can't executed exchange as he don't know how much
			// amount of token he has to trade-in to get expected buy tokens.
			// So we compute price assuming user wants to sell instead of buy.
			// And then do exchange computed amount with token indices flipped.
			let dx = Self::get_exchange_value(pool_id, asset_id, amount)?;
			Self::update_balance(who, &pool_id, &exchange_asset, dx, &asset_id, amount)?;
			Ok(amount)
		}
```
The implementation of `buy` is such that it asks `dx` amount from the user which is required to buy `amount` tokens so its not required to have slippage.

Reply:
nice documentation. could we have that on pallet call level or other means in /// code?